docs/_templates/search.html

```diff
@@ -32,13 +46,85 @@
+                    displayPagination(page, nbPages)
+                }
+
+                // Display hits
+                let hits = result["hits"]
+                addResultsList(hits, query, searchResultsContainer)
+            })
                 .catch((error) => console.log(error))
         }
 
+        function displayPagination(page, totalPages) {
+            const pagination = document.querySelector("#pagination")
+            const paginationList = pagination.querySelector(".pagination-list")
+
+            // Hide previous/next button if showing first/last page
+            pagination.querySelector(".pagination-previous").hidden= false;
+            pagination.querySelector(".pagination-previous").hidden = page == 0;
+            pagination.querySelector(".pagination-next").hidden = page == totalPages -1;
+
+            // Display at most "toBeDisplayed" page links in the paginator
+            const toBeDisplayed = 7
+            let [start, end] = setStartEndForPaginator(page, totalPages, toBeDisplayed)
+
+            for (let i = start; i < end; i++) {
+                let newPaginationItem = document.createElement("li")
+                let newPaginationbutton = document.createElement("button")
+                newPaginationbutton.classList.add("pagination-button")
+                newPaginationbutton.innerHTML = i + 1
+                if (i == page) {
+                    newPaginationbutton.setAttribute("aria-label", `page ${i+1}`)
+                    newPaginationbutton.setAttribute("aria-current", `page`)
+                } else {
+                    newPaginationbutton.setAttribute("aria-label", `Go to page ${i+1}`)
+                }
+
+                // Register event handlers
+                newPaginationbutton.onclick = function(event) {
+                    runSearchPageSearch(event, page=i)
+                };
+                pagination.querySelector(".pagination-previous").onclick = function(event) {
+                    runSearchPageSearch(event, page=page - 1)
+                };
+                pagination.querySelector(".pagination-next").onclick = function(event) {
+                    runSearchPageSearch(event, page=page + 1)
+                };
+
+                newPaginationItem.append(newPaginationbutton)
+                paginationList.append(newPaginationItem)
+            }
+        }
+
+        function setStartEndForPaginator(currentPage, totalPages, nbPagesLinkToDisplay) {
+            let start = 0
+            let end = totalPages
+
+            if (totalPages > nbPagesLinkToDisplay) {
+                start = currentPage
+
+                if (totalPages - currentPage < nbPagesLinkToDisplay) {
+                    start = totalPages - nbPagesLinkToDisplay
+                }
+
+                if (totalPages - start > nbPagesLinkToDisplay) {
+                    end = currentPage + nbPagesLinkToDisplay
+                }
+            }
+
+            return [start, end]
+        }
+
```

Comment:
Just a thought, is it worth bringing this code out to its own JS file.

Reply:
@PaarthAgarwal - I think it would be good to move this whole script block to an external file. 

Similar to the way that `banner.js` is done. 
https://github.com/wagtail/wagtail/blob/main/docs/_static/js/banner.js

Maybe let's call this `docs.js`

You will need to ensure the file gets included in conf.py 
https://github.com/wagtail/wagtail/blob/1a49cae272f9fd5407bff3d97b7a1dfd5af9b122/docs/conf.py#L299